src/index.ts

```diff
@@ -854,6 +870,64 @@ export abstract class PinejsClientCore<PinejsClient> {
 		}
 	}
 
+	private canRetryDefaultHandler(err: any) {
+		const code = err?.statusCode;
+		return !code || code === 429 || (code >= 500 && code < 600);
+	}
+
+	protected async callWithRetry<T>(
+		fnCall: () => Promise<T>,
+		retry: RetryParameters,
+	): Promise<T> {
+		if (!retry) {
+			return await fnCall();
+		}
+
+		const defaultRetry = this.retry || {};
+		const minDelayMs = retry.minDelayMs ?? defaultRetry.minDelayMs ?? 0;
+		const maxDelayMs = retry.maxDelayMs ?? defaultRetry.maxDelayMs ?? 0;
+		const maxAttempts = retry.maxAttempts ?? defaultRetry.maxAttempts ?? 0;
+
+		if (
+			minDelayMs <= 0 ||
+			maxDelayMs <= 0 ||
+			maxAttempts <= 0 ||
+			minDelayMs > maxDelayMs
+		) {
+			console.warn('pinejs-client invalid retry parameters');
```

Comment:
I'd rather this throws so that it's immediately obvious that it won't do what the programmer intends rather than silently sort of working

Reply:
Are you super sure?  What if there are some defaults that are used at construction, then the caller overrides one of them in some cases and accidentally ends up with invalid parameters?  To me, the retry code should never _cause_ a failure that would not otherwise be there.  I'll go with your judgement on the matter, but to me this feels like it could bite us.

Reply:
My thoughts on it is that if someone is passing invalid parameters then it means they have a bug in their code and as such should fix it, immediately throwing means it's easy to see where is causing it and fix it (presumably during testing, since I wouldn't expect to provide dynamic values that regularly change from call to call - the retrying code itself is what's meant to do the dynamic values part)